### PR TITLE
DOC/REF: Clarify pip extras dependencies & cleanups

### PR DESCRIPTION
**Pull request by mroeschke**
Clarifying that https://github.com/pandas-dev/pandas/pull/47336 is specifically for managing `pip` installation of optional dependencies

### CODE REVIEW

File: doc/source/getting_started/install.rst [hunk 1]
```diff
@@ -139,6 +139,16 @@ pandas can be installed via pip from
 
     pip install pandas
 
+pandas can also be installed with sets of optional dependencies to enable certain functionality. For example,
+to install pandas with the optional dependencies to read Excel files.
+
+::
+
+    pip install pandas[excel]
```
Comment (MarcoGorelli) [inline doc/source/getting_started/install.rst:147]: I think this won't work on mac,
```suggestion
    pip install "pandas[excel]"
```
should work everywhere

(likewise for the others)

https://github.com/psf/black/pull/3254#issuecomment-1236197864

**Review by MarcoGorelli — COMMENTED** [review]
Generally looks good, just left a comment about making this cross-shell

File: doc/source/getting_started/install.rst (hunk 1 above)
Comment (mroeschke) [reply doc/source/getting_started/install.rst]: Good idea! Added

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me, thanks @mroeschke !

**Pull request merged**